src/std/misc/decimal-test.ss

```diff
@@ -6,58 +6,114 @@
   :std/misc/decimal
   :std/misc/string
   :std/parser/base
-  :std/test)
+  (only-in :std/error ContractViolation?)
+  (only-in :std/sugar defrule)
+  (only-in :std/text/char-set char-ascii-alphabetic?)
+  (only-in :std/test test-suite test-case check check-exception))
 
 (def decimal-test
   (test-suite "test suite for std/misc/decimal"
+    (test-case "power-of-5?"
+      (for (i (in-range 1000))
+        (check (power-of-5? (expt 5 i)) => i)
+        (check (power-of-5? (1- (expt 5 i))) => #f)
+        (check (power-of-5? (1+ (expt 5 i))) => #f))
+      (check (power-of-5? (* (expt 2 2000) (expt 5 880))) => #f)
+      (check (power-of-5? (* 256 (expt 5 4400))) => #f))
+    (test-case "decimal?"
+      (defrule (checks res val ...)
+        (begin (check (decimal? val) => res) ...))
```

Comment:
why does this have a `?` in the name? Doesn't look like it is a predicate.

Reply:
Fixed.